app/components/buttons/EyeFilterMenuWithSlider/index.js

```diff
@@ -0,0 +1,152 @@
+import { EyeFilterMenu } from "buttons";
+import { FormattedMessage as T } from "react-intl";
+import "style/EyeFilterMenu.less";
+import "style/MiscComponents.less";
+import noUiSlider from "nouislider";
+import {  NumericInput } from "inputs";
+
+@autobind
+class EyeFilterMenuWithSlider extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      minAmount: 0,
+      maxAmount: 100,
+      min: 0,
+      max: 100,
+      sliderShower: true,
+      expandedSliderInfo: false,
+    };
+  }
+
+  getSliderWhenOpeningMenu() {
+    setTimeout(() => {
+      const range = document.getElementById("min-max-slider");
```

Comment:
Don't use `getElementById` as it breaks componentization (you wouldn't be able to add 2 instances of `EyeFilterMenuWithSlider` at the same time due to them having the same id).

Parametrize `getSliderWhenOpeningMenu` (renaming it to `createSliderRangeInElement`) with what you used as `range` (that's the `range` parameter), then add a `ref` prop to `#min-max-slider` (I'll not where below).

Reply:
good catch